### ISSUE TITLE
ProcessOperationsAsync should check if operations were already processed before

### ISSUE THREAD

**Issue by Toccata6767** [stefffdev/NubeSync#23]
It could be that I'm doing something wrong but I've been messing with this a lot recently and I'm pretty sure it's a bug/flaw. There's an issue that occurs when the client pushes changes to the server to process, the server successfully processes them, and then for whatever reason the client doesn't receive the Ok http status code back from the server. Could be due to anything, intermittent connections, crashes, etc, doesn't really matter, but it's a scenario that needs to be considered. 

Normally the client waits to receive the Ok status code and then deletes those operations from the sqlite db. In this case the client never receives the Ok status code back and so the operations never get deleted from the sqlite db. The next time the client tries to push changes, the previous operations that the sever has already processed are included in that list. ProcessOperationsAsync doesn't seem to filter out operations that have already been processed, so it attempts to process them anyway. An exception will get thrown when context.SaveChangesAsync() is called for attempting to insert duplicate primary keys into the sql db and no data will be saved. The server sends back a BadRequest status code, so those operations that are causing problems remain on the client. Now this client is stuck in a state where it can't sync data with the server.

I was able to resolve this by placing these lines of code in NubeSync.Server/OperationService.cs at the top of the ProcessOperationsAsync method.

```C#
var operationIds = operations.Select(o => o.Id);
var alreadyProcessed = context.Set<NubeServerOperation>().Where(o => operationIds.Contains(o.Id));
operations = operations.Where(o => !alreadyProcessed.Any(x => x.Id == o.Id)).ToList();
```

This will basically force the server to check for operations that have already been processed and then filter them out of the incoming list of operations. Eventually when the client does receive it's Ok status code those operations get removed and it goes back to normal, so the server doesn't have to filter those out every time. Not sure if this is the best or most efficient way of handling this but maybe it will at least give you an idea.

**Comment by stefffdev** [comment]
Hi, you are right with your findings.
Internally we handle this case on the client, be removing those operations manually when that certain BadRequest is returned.

But I agree that this is something that should be handled by the server by not processing these operations any more.
Will try to fix that soon, since it only should be a small fix!

**Comment by Toccata6767** [comment]
Awesome, thank you for your quick response. And thank you for making this project available to everyone. It has been a huge help for my project.

**Comment by stefffdev** [comment]
Thanks, I published a pre-release on nuget (1.5.0-pre2) that should fix that issue.
